nixos/modules/services/web-apps/flarum.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  inherit (lib) mkDefault mkEnableOption mkIf mkOption types literalExample;
+  cfg = config.services.flarum;
+in {
+  options = {
+    services.flarum = {
+      enable = mkEnableOption "Flarum discussion platform";
+
+      packages = mkOption {
+        type = with types; attrsOf package;
+        example = literalExample "pkgs.flarum";
+        description = "Packages used for the service.";
+        default = pkgs.flarum;
+      };
+
+      domain = mkOption {
+        type = types.str;
+        default = "localhost";
+        example = "forum.example.com";
+        description = "Domain to serve on.";
+      };
+
+      installPath = mkOption {
+        type = types.str;
+        default = "/var/www/flarum";
+        description = "Installation location in the file system";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "nobody";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "nogroup";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.phpfpm.pools.flarum = {
+      user = cfg.user;
+      settings = {
+        "listen.owner" = config.services.nginx.user;
+        "listen.mode" = "0600";
+        "pm" = mkDefault "dynamic";
+        "pm.max_children" = mkDefault 10;
+        "pm.max_requests" = mkDefault 500;
+        "pm.start_servers" = mkDefault 2;
+        "pm.min_spare_servers" = mkDefault 1;
+        "pm.max_spare_servers" = mkDefault 3;
+      };
+    };
+
+    services.nginx = {
+      enable = true;
+      virtualHosts."${cfg.domain}" = {
+        root = "${cfg.installPath}/public";
+        locations."~ \.php$".extraConfig =
+          ''
+          fastcgi_pass unix:${config.services.phpfpm.pools.flarum.socket};
+          fastcgi_index site.php;
+          '';
+        extraConfig =
+          ''
+          index index.php;
+          include ${cfg.installPath}/.nginx.conf;
+          '';
+      };
+    };
+
+    systemd.services.flarum-install = {
+      description = "Flarum installation";
+      before = [ "nginx.service" "phpfm-flarum.service" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "oneshot";
+      };
+      script = with cfg; ''
```

Comment:
It might be worth replacing this with `systemd.tmpfiles.rules`.

Reply:
The installPath will contain state: `config.php`, `storage/`. The software checks for writeability of these paths.

Reply:
@astro can this be resolved after doing https://github.com/NixOS/nixpkgs/pull/96869#discussion_r482613392 ?